glib-macros/src/props.rs

```diff
@@ -0,0 +1,498 @@
+        quote!(
+            #getter
+            #setter
+        )
+    });
+    quote!(#(#defs)*)
+}
+
+fn expand_getset_properties_impl(imp_type_ident: &syn::Ident, props: &[PropDesc]) -> TokenStream2 {
+    let defs = props.iter().map(|p| {
+        let ident = name_to_ident(&p.name);
+        let set_ident = format_ident!("set_{}", ident);
+        let field_ident = &p.field_ident;
+        let ty = &p.ty;
+
+        let getter = p.get.as_ref().map(|mfn| {
+            let body = match (p.member.as_ref(), mfn) {
+                (None, MaybeCustomFn::Default) => quote!(
+                     self.imp().#field_ident.get(|x| x.to_owned())
+                ),
+                (Some(member), MaybeCustomFn::Default) => quote!(
+                     self.imp().#field_ident.get(|x| x.#member.to_owned())
+                ),
+                (_, MaybeCustomFn::Custom(custom_fn)) => {
+                    let custom_fn = change_self_type(custom_fn, &imp_type_ident.to_string());
+                    quote!((#custom_fn)(&self.imp()))
+                }
+            };
+            quote!(fn #ident(&self) -> <#ty as glib::Property>::Inner {
+                #body
+            })
+        });
+        let setter = p.set.as_ref().map(|mfn| {
+            let body = match (p.member.as_ref(), mfn) {
+                (None, MaybeCustomFn::Default) => quote!(
+                     self.imp().#field_ident.set(move |x| *x = value)
+                ),
+                (Some(member), MaybeCustomFn::Default) => quote!(
+                     self.imp().#field_ident.set(move |x| x.#member = value)
+                ),
+                (_, MaybeCustomFn::Custom(custom_fn)) => {
+                    let custom_fn = change_self_type(custom_fn, &imp_type_ident.to_string());
+                    quote!((#custom_fn)(&self.imp(), value))
+                }
+            };
+            quote!(fn #set_ident(&self, value: <#ty as glib::Property>::Inner) {
+                #body
+            })
+        });
+        quote!(
+            #getter
+            #setter
+        )
+    });
+    quote!(#(#defs)*)
+}
+
+pub fn impl_derive_props(input: PropsMacroInput) -> TokenStream {
+    let struct_ident = &input.ident;
+    let struct_ident_ext = format_ident!("{}Ext", &input.ident);
+    let wrapper_type = quote!(<#struct_ident as glib::subclass::types::ObjectSubclass>::Type);
+    let fn_properties = expand_properties_fn(&input.props);
+    let fn_property = expand_property_fn(&input.props);
+    let fn_set_property = expand_set_property_fn(&input.props);
+    let getset_properties_def = expand_getset_properties_def(&input.props);
+    let getset_properties_impl = expand_getset_properties_impl(struct_ident, &input.props);
+    let expanded = quote! {
+        use glib::{PropRead, PropWrite};
+        impl ObjectImpl for #struct_ident {
+            #fn_properties
+            #fn_property
+            #fn_set_property
+        }
+
+        pub trait #struct_ident_ext {
```

Comment:
Should also generate a connect_$prop_notify

Reply:
But gtk4-rs doesn't have `connect_$prop_notify` methods...

Reply:
That seems like a problem, can you create an issue about that?